File: unidad2/Employee.java
import java.util.ArrayList;

public class Employee {
    private String name;
    private String lastName;
    private BankAccount account;
    private boolean status = true;
    private ArrayList <BankAccount> AccountList = new ArrayList<>();

    public Employee(String name, String lastName) {
        this.name = name;
        this.lastName = lastName;
    }

    public Employee(String name, String lastName, long accountNumber, String type) {
        if(validType(type)){
            this.name = name;
            this.lastName = lastName;
            this.account = new BankAccount(accountNumber, type);
            AccountList.add(account);
        }
        else{
            this.status = false;
        } 
    }

    public String getName() {
        return name;
    }

    public String getLastName() {
        return lastName;
    }

    public BankAccount getAccount() {
        return account;
    }

    public boolean isStatus() {
        return status;
    }

    public ArrayList <BankAccount> getAccountList(){//CREO QUE NO LO OCUPAS    
        return AccountList;
    }

    public BankAccount addAccount(long accountNumber, String type){
        BankAccount newAccount = null;
        if(validType(type)){
            newAccount = new BankAccount(accountNumber, type);
            AccountList.add(newAccount);
        }
        return newAccount;
    }

    private boolean validType(String type){
        switch(type){
            case "A":
            case "B":
            case "C":
                return true;

            default:
                return false;
        }
    }

    public String getFullName(){
        return name + " " + lastName;
    }

    /*private String generarID(){
        String bank = "abcdefghijklmnñopqrstuvwxyzABCDEFGHIJKLMNÑOPQRSTUVWXYZ1234567890";
        String cad = "";
        for (int i = 0; i < 5; i++) {
            cad += bank.charAt(ran.nextInt(bank.length()));
        }
        return cad;
    }
    */
}

File: unidad2/Employee.py
import BankAccount

class Employee:
    __name = ''
    __last_name = ''
    __account = BankAccount()
    __status = True
    __account_list = []

    def __init__(self, name, last_name, account_number, type):
        if(self._valid_type(type)):
            self.__name = name
            self.__last_name = last_name
            self.__account = BankAccount(account_number, type)
            __account_list.append(self.__account)
        else:
            self.__status = False

    def get_name(self):
        return self.__name
    
    def get_last_name(self):
        return self.__last_name
    
    def get_account(self):
        return self.__account
    
    def is_status(self):
        return self.__status
    
    def get_account_list(self):
        return __account_list
    
    def add_account(self, account_number, type):
    ##VAS EN LA PARTE QUE HACE NULL el objeto newAccount Edgar
    
        


    def _valid_type(self, type):
        if type == 'A':
            return True
        elif type == 'B':
            return True
        elif type == 'C':
            return True
        else:
            return False
        
